templates/pkg/resource/sdk.go.tpl

```diff
@@ -218,16 +219,16 @@ func (rm *resourceManager) updateConditions (
 			syncCondition = condition
 		}
 	}
-
-	if rm.terminalAWSError(err) || err ==  ackerr.SecretTypeNotSupported || err == ackerr.SecretNotFound {
+	var termError ackerr.TerminalError
+	if rm.terminalAWSError(err) || err ==  ackerr.SecretTypeNotSupported || err == ackerr.SecretNotFound || errors.As(err, &termError) {
```

Comment:
Instead, use `if errors.Is(err, ackerr.TerminalError)` :)

(Honestly these other error checks should also be using that syntax)

Reply:
Oops. Thank you for bringing to my attention that `errors.Is` works on value comparison, and not type assertion. `errors.As` is the proper use here.